### DISCUSSION TITLE
How to correctly set endpoints for static files?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#2568] [Q&A] [answered]
### Question

Hi All,

I am trying to declare a folder as an endpoint so that I can use some images from there in other parts of the code. But seems like I am getting some error in creating the url. Below is what I am testing:
```python
from nicegui import ui,app
import os, sys

images = os.path.join(sys.prefix, 'app', 'images')    # this is my folder that I want to access from url
app.add_static_files('/images', images)

ui.label('This is a test')
ui.button('Images', on_click=lambda: ui.open('/images'))

ui.run(port=5002)
```

But if I press the button, or manually go to the '\images' url, no the screen it just says 'disconnected' like this below
![image](https://github.com/zauberzeug/nicegui/assets/65127974/43067184-d8f7-4bc1-b3e5-5e84ec9e2b2f)

And on the terminal, I get few other error lines like this below
```text
http://localhost:5002/images/ not found
http://localhost:5002/images/_nicegui/1.4.15/static/nicegui.css not found
http://localhost:5002/images/_nicegui/1.4.15/static/fonts.css not found
http://localhost:5002/images/_nicegui/1.4.15/static/es-module-shims.js not found
http://localhost:5002/images/_nicegui/1.4.15/static/quasar.prod.css not found
http://localhost:5002/images/_nicegui/1.4.15/static/socket.io.min.js not found
http://localhost:5002/images/_nicegui/1.4.15/static/tailwindcss.min.js not found
http://localhost:5002/images/_nicegui/1.4.15/static/vue.global.prod.js not found
http://localhost:5002/images/_nicegui/1.4.15/static/quasar.umd.prod.js not found
http://localhost:5002/images/_nicegui/1.4.15/static/lang/en-US.umd.prod.js not found
```

I am unable to understand the error. Can you please guide me on this, as to how to correctly declare the folder as an endpoint url?

Thanks a lot,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088,

The root cause of the page not being shown is that the `StaticFiles` route from FastAPI doesn't list directories. Visiting individual images should work as expected.

But instead of getting lots of 404 errors, you should see a "nice" error page. There seems to be a bug when routing its assets, which I created issue #2570 for.

**Reply by Anindya088** [reply]
Hi @falkoschindler , 

Thanks a lot for explaining, so the folder itself will not be accessible, just individual files, understood now.
So is there any other way to make a full folder accessible like a directory which then I can browse from the front end?

Thanks a lot,
Anindya

**Reply by rodja** [reply]
Have a look at our [local file picker example](https://github.com/zauberzeug/nicegui/tree/main/examples/local_file_picker).

**Reply by Anindya088** [reply]
Hi @rodja , oh this is great, sorry should have searched first before asking. Thank you so much 😄
